dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkflowAPIImpl.java

```diff
@@ -944,8 +965,14 @@ public WorkflowHistory retrieveLastStepAction(String taskId) throws DotDataExcep
 
 	@CloseDBIfOpened
 	public WorkflowAction findEntryAction(Contentlet contentlet, User user)  throws DotDataException, DotSecurityException {
-
-		WorkflowScheme scheme = findSchemeForStruct(contentlet.getStructure());
+		WorkflowScheme scheme = null;
+		List<WorkflowScheme> schemes = findSchemeForStruct(contentlet.getStructure());
+		if(null !=  schemes && schemes.size() ==1){
+			scheme =  schemes.get(0);
+		}else{
+			return null;
+		}
+		/* END - TODO this part need to be done in a different way */

```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)